vsphere/datadog_checks/vsphere/api.py

```diff
@@ -61,21 +67,27 @@ class VSphereAPI(object):
     """Abstraction class over the vSphere SOAP api using the pyvmomi library"""
 
     def __init__(self, config, log):
+        # type: (VSphereConfig, CheckLoggingAdapter) -> None
         self.config = config
         self.log = log
 
-        self._conn = None
+        self._conn = cast(vim.ServiceInstance, None)
```

Comment:
Didn't tick on this one, but probably worth converting to a property too:

```python
    @property
    def _conn(self):
        # type: () -> vim.ServiceInstance
        if not hasattr(self, '__conn'):
            raise RuntimeError('Not connected yet')
        return self.__conn

    @_conn.setter
    def _conn(self, conn):
        # type: (vim.ServiceInstance) -> None
        existing_conn = getattr(self, '__conn', None)
        if existing_conn is not None:
            connect.Disconnect(existing_conn)
        self.__conn = conn
```

Reply:
I think that's adding a lot of complexity and would prefer to keep it simple. Some changes are ok to satisfy typing rules, but this seems a bit overkill.